src/java/org/apache/sqoop/mapreduce/mainframe/MainframeDatasetBinaryRecord.java

```diff
@@ -0,0 +1,128 @@
+/**
```

Comment:
I am still not sure we will need this new class.
I understand that when a SqoopRecord is generated runtime it will always have a text field and not binary but that might be solved by changing org.apache.sqoop.manager.MainframeManager#getColumnTypes

My guess is that if we changed that to return column type based on the transfer mode or file type in the SqoopOptions object we could just use the generated class.